src/main/java/com/teragrep/rlp_11/RelpProbeConfiguration.java

```diff
@@ -0,0 +1,169 @@
+
+public class RelpProbeConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RelpProbeConfiguration.class);
+    private static final int MAXIMUM_PORT = 65535;
+    private final Map<String, String> config;
+
+    public RelpProbeConfiguration(final Map<String, String> config) {
+        this.config = config;
+    }
+
+    public void validate() throws RelpProbeConfigurationError {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Validating the following configuration: {}", config);
+        }
+
+        getAndVerifyStringProperty("event.hostname");
+        getAndVerifyStringProperty("event.appname");
+
+        int eventDelay = getAndVerifyNumberProperty("event.delay");
+        if (eventDelay <= 0) {
+            LOGGER.error("Invalid <event.delay> property, expected > 0, received <[{}]>", eventDelay);
+            throw new RelpProbeConfigurationError(
+                    "Invalid <event.delay> property, expected > 0, received <[" + eventDelay + "]>"
+            );
+        }
+
+        getAndVerifyStringProperty("target.hostname");
+
+        int targetPort = getAndVerifyNumberProperty("target.port");
+        if (targetPort <= 0 || targetPort > MAXIMUM_PORT) {
+            LOGGER
+                    .error(
+                            "Invalid <target.port> property, expected between 1 and {}, received <[{}]>", MAXIMUM_PORT,
+                            targetPort
+                    );
+            throw new RelpProbeConfigurationError(
+                    "Invalid <target.port> property, expected between 1 and " + MAXIMUM_PORT + ", received <["
+                            + targetPort + "]>"
+            );
+        }
+
+        int prometheusPort = getAndVerifyNumberProperty("prometheus.port");
+        if (prometheusPort <= 0 || prometheusPort > MAXIMUM_PORT) {
+            LOGGER
+                    .error(
+                            "Invalid <prometheus.port> property, expected between 1 and {}, received <[{}]>",
+                            MAXIMUM_PORT, prometheusPort
+                    );
+            throw new RelpProbeConfigurationError(
+                    "Invalid <prometheus.port> property, expected between 1 and " + MAXIMUM_PORT + ", received <["
+                            + prometheusPort + "]>"
+            );
+        }
+
+        int reconnectInterval = getAndVerifyNumberProperty("target.reconnectinterval");
+        if (reconnectInterval <= 0) {
+            LOGGER
+                    .error(
+                            "Invalid <target.reconnectinterval> property, property, expected > 0, received <[{}]>",
+                            reconnectInterval
+                    );
+            throw new RelpProbeConfigurationError(
+                    "Invalid <target.reconnectinterval> property, expected > 0, received <[" + reconnectInterval + "]>"
+            );
+        }
+    }
+
+    private String getAndVerifyStringProperty(String name) {
```

Comment:
Not the best naming/practices, maybe should do something more elegant with these